test/PurchasabilityValidator.test.js

```diff
@@ -0,0 +1,175 @@
+    // given
+    const drugItem = await DrugItem.new(drugItemIdBytes, vendor.id, pharmacy.id, pharmacy.category);
+    // when
+    const actualCodes = await sut.isPurchasable(drugItem.address);
+    // then
+    expectPurchasabilityCodes(actualCodes).toEqual([purchasabilityCodes.ValidForPurchase]);
+  });
+
+  it('should return error code if the drug item is not in a pharmacy', async () => {
+    // given
+    const drugItem = await DrugItem.new(drugItemIdBytes, vendor.id, carrier1.id, carrier1.category);
+    // when
+    const actualCodes = await sut.isPurchasable(drugItem.address);
+    // then
+    expectPurchasabilityCodes(actualCodes).toEqual([purchasabilityCodes.NotInPharmacy]);
+  });
+
+  it('should return error code if the drug item was handed over by carriers more than twice', async () => {
+    // given
+    const drugItem = await DrugItem.new(drugItemIdBytes, vendor.id, carrier1.id, carrier1.category);
+    await logHandoverFromCarrier(drugItem, carrier1, carrier2);
+    await logHandoverFromCarrier(drugItem, carrier2, carrier1);
+    await logHandoverFromCarrier(drugItem, carrier1, carrier2);
+    await logHandoverFromCarrier(drugItem, carrier2, pharmacy);
+    // when
+    const actualCodes = await sut.isPurchasable(drugItem.address);
+    // then
+    expectPurchasabilityCodes(actualCodes).toEqual([purchasabilityCodes.TooManyHandovers]);
+  });
+
+  const carrierWithTemperature = (category, temperature) => {
+    const carrier = participants.carrier(accounts[9], category);
+    carrier.conditions.temperature = temperature;
+    return carrier;
+  };
+
+  const carrierTooHighTemperatureTestCases = [{
+    name: 'ship temperature was above -18 degrees',
+    carrier: carrierWithTemperature(carrierCategories.Ship, -17),
+  }, {
+    name: 'airplace temperature was above -10 degrees',
+    carrier: carrierWithTemperature(carrierCategories.Airplane, -9),
+  }, {
+    name: 'truck temperature was above -18 degrees',
+    carrier: carrierWithTemperature(carrierCategories.Truck, -17),
+  }];
+
+  carrierTooHighTemperatureTestCases.forEach(({ name, carrier }) => {
+    it(`should return error code if ${name}`, async () => {
+      // given
+      const drugItem = await DrugItem.new(drugItemIdBytes, vendor.id, carrier1.id, carrier1.category);
+      await logHandoverFromCarrier(drugItem, carrier1, carrier);
+      await logHandoverFromCarrier(drugItem, carrier, pharmacy);
+      // when
+      const actualCodes = await sut.isPurchasable(drugItem.address);
+      // then
+      expectPurchasabilityCodes(actualCodes).toEqual([purchasabilityCodes.TemperatureTooHigh]);
+    });
+  });
+
+  it('should return error code if temperature dropped below -22 degrees', async () => {
+    // given
+    const drugItem = await DrugItem.new(drugItemIdBytes, vendor.id, carrier1.id, carrier1.category);
+    const carrier = carrierWithTemperature(carrierCategories.Ship, -23);
```

Comment:
magic number in variable will be less confusing

Reply:
I'm not so sure about that - you see a `carrierWithTemperature()` with plainly a carrier category and the temperature. Additionally you have the context of the test name. Extracting a variable such as `temperatureBelowLimit` would add a bit more noise. But that's my opinion, was your perception of the code different from mine?

Reply:
In my opinion this `carrierWithTemperature(carrierCategories.Ship, temperatureBelowLimit);` looks more readable than `carrierWithTemperature(carrierCategories.Ship, -23);` but is not such important thing.